redash/handlers/users.py

```diff
@@ -149,6 +149,27 @@ def post(self, user_id):
         }
 
 
+class UserRegenerateApiKeyResource(BaseResource):
+    def post(self, user_id):
+        user = models.User.get_by_id_and_org(user_id, self.current_org)
+        if user.is_disabled:
+            abort(404, message='Not found')
+        if not is_admin_or_owner(user_id):
+            abort(403)
+
+        user.regenerate_api_key()
+        models.db.session.commit()
+
+        self.record_event({
+            'action': 'edit',
+            'object_id': user.id,
+            'object_type': 'user',
+            'updated_fields': ['api_key']
+        })
```

Comment:
I would use `regnerate_api_key` as `action` and remove the `updated_fields` field.